actions/class.ItemContent.php

```diff
@@ -202,7 +203,13 @@ private function getMediaAsset(bool $validateWriteAccess = false): ?MediaAsset
     {
         $params = $this->getRequiredQueryParams('uri', 'lang', 'path');
         $asset = $this->resolveAsset($params['uri'], $params['path'], $params['lang']);
-        $resourceUri = tao_helpers_Uri::decode($asset->getMediaIdentifier());
+
+        $mediaSource = $asset->getMediaSource();
+        if ($mediaSource instanceof LocalItemSource) {
+            $resourceUri = $params['uri'];
+        } else {
+            $resourceUri = tao_helpers_Uri::decode($asset->getMediaIdentifier());
+        }
```

Comment:
We can simplify the code:

```suggestion
        $resourceUri = $asset->getMediaSource() instanceof LocalItemSource) 
            ? $params['uri']
            : tao_helpers_Uri::decode($asset->getMediaIdentifier());
```

Reply:
Done